### PR TITLE
Fixed module preload in building-utils.

### PR DESCRIPTION
**Pull request by soelen**
## What I did

image

1. main script was request 2 times when built
2. Investigated a bit, asked team open-wc on slack
3. Investigated more, found this neat [Article](https://developers.google.com/web/updates/2017/12/modulepreload)
4. Searched code snippet where link element is actually being created
5. Created PR

### CODE REVIEW

**Comment by Westbrook** [comment]
I think the remaining test failures are "expected". If you run the following script, we should be full green here: https://github.com/soelen/open-wc/blob/building-utils-module-fix/packages/building-utils/package.json#L19

**Comment by soelen** [comment]
Unfortunatly running `test:update-snapshots` did not the trick

**Comment by stephenwade** [comment]
The latest failures are hash changes to the core-js polyfill, which is confusing because that wasn't updated in this PR.

**Comment by soelen** [comment]
Mhhhh, perhaps I shouldnt have commited some of the snapshots, tomorrow I will give another try to find out what is going on :)

**Comment by Westbrook** [comment]
Interesting. My pass on this in #2121 doesn't seem to have he same CoreJS changes.

@soelen can you share your OS info? Also, you _might_ be able to get back to the same output if you `rm -rf node_modules && yarn install`?

**Comment by soelen** [comment]
> Interesting. My pass on this in #2121 doesn't seem to have he same CoreJS changes.
> 
> @soelen can you share your OS info? Also, you _might_ be able to get back to the same output if you `rm -rf node_modules && yarn install`?

Windows 10, WSL -> Ubuntu 18.04.4 LTS

Tried it with yarn on different ways but it seems like nothing works as supposed to; with or without WSL :(

File: packages/building-utils/test-node/index-html/snapshots/create-index-html/polyfill-multiple-legacy.html [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<html lang="en-GB"><head><title>My app</title><style>my-app{display:block}</style><link rel="modulepreload" href="app.js" as="script"></head><body><h1><span>Hello world!</span></h1><my-app></my-app><script src="polyfills/core-js.965e31c81eb8176876bb7433a6f161ab.js" nomodule=""></script><script>!function(){function e(e,o){return new Promise((function(t,n){document.head.appendChild(Object.assign(document.createElement("script"),{src:e,onload:t,onerror:n},o?{type:"module"}:void 0))}))}var o=[];function t(){"noModule"in HTMLScriptElement.prototype?window.importShim("./app.js"):System.import("./legacy/app.js")}(!("attachShadow"in Element.prototype)||!("getRootNode"in Element.prototype)||window.ShadyDOM&&window.ShadyDOM.force)&&o.push(e("polyfills/webcomponents.f2956532e796a677ef87a9cb17f5f9fc.js",!1)),!("noModule"in HTMLScriptElement.prototype)&&"getRootNode"in Element.prototype&&o.push(e("polyfills/custom-elements-es5-adapter.84b300ee818dce8b351c7cc7c100bcf7.js",!1)),o.length?Promise.all(o).then(t):t()}()</script></body></html>
```
Comment (stephenwade) [inline packages/building-utils/test-node/index-html/snapshots/create-index-html/polyfill-multiple-legacy.html:1]:
```suggestion
<html lang="en-GB"><head><title>My app</title><style>my-app{display:block}</style><link rel="modulepreload" href="app.js" as="script"></head><body><h1><span>Hello world!</span></h1><my-app></my-app><script src="polyfills/core-js.271cd0c5fd1a210a2c430d4d42c6ea72.js" nomodule=""></script><script>!function(){function e(e,o){return new Promise((function(t,n){document.head.appendChild(Object.assign(document.createElement("script"),{src:e,onload:t,onerror:n},o?{type:"module"}:void 0))}))}var o=[];function t(){"noModule"in HTMLScriptElement.prototype?window.importShim("./app.js"):System.import("./legacy/app.js")}(!("attachShadow"in Element.prototype)||!("getRootNode"in Element.prototype)||window.ShadyDOM&&window.ShadyDOM.force)&&o.push(e("polyfills/webcomponents.f2956532e796a677ef87a9cb17f5f9fc.js",!1)),!("noModule"in HTMLScriptElement.prototype)&&"getRootNode"in Element.prototype&&o.push(e("polyfills/custom-elements-es5-adapter.84b300ee818dce8b351c7cc7c100bcf7.js",!1)),o.length?Promise.all(o).then(t):t()}()</script></body></html>
```

File: packages/building-utils/test-node/index-html/snapshots/create-index-html/polyfill-multiple.html [hunk 2]
```diff
@@ -1 +1 @@
...
...ment.baseURI):t)</script><script src="polyfills/custom-b.36a50cce88edee34c249d0276be6531d.js" nomodule=""></script><script src="polyfills/core-js.965e31c81eb8176876bb7433a6f161ab.js" nomodule=""></script><script>!function(){function e(e,o){return new Promise((function(t,n){document.head.appendChild(Object.assign(document.createElement("script"),{src:e,onload:t,onerror:n},o?{type:"module"}:void 0))}))}var o=[];function t(){window.importShim("./app.js")}"foo"in window&&o.push(e("polyfills/custom-a.612310cce7c28a680112cc9eff6ef77c.js",!1)),"fetch"in window||o.push(e("polyfills/fetch.a1ad5fb96dc0cb61b9454244c9bd7fe6.js",!1)),"noModule"in HTMLScriptElement.prototype&&!("importShim"in window)&&o.push(e("polyfills/dynamic-import.991be47e17117abb5eb15f5254ad3869.js",!1)),(!("attachShadow"in Element.prototype)||!("getRootNode"in Element.prototype)||window.ShadyDOM&&window.ShadyDOM.force)&&o.push(e("polyfills/webcomponents.f2956532e796a677ef87a9cb17f5f9fc.js",!1)),!("noModule"in HTMLScriptElement.prototype)&&"getRootNode"in Element.prototype&&o.push(e("polyfills/custom-elements-es5-adapter.84b300ee818dce8b351c7cc7c100bcf7.js",!1)),o.length?Promise.all(o).then(t):t()}()</script></body></html>
```
Comment (stephenwade) [inline packages/building-utils/test-node/index-html/snapshots/create-index-html/polyfill-multiple.html:1]:
```suggestion
<html lang="en-GB"><head><title>My app</title><style>my-app{display:block}</style><link rel="modulepreload" href="app.js" as="script"></head><body><h1><span>Hello world!</span></h1><my-app></my-app><script>window.importShim=t=>import(t.startsWith(".")?new URL(t,document.baseURI):t)</script><script src="polyfills/custom-b.36a50cce88edee34c249d0276be6531d.js" nomodule=""></script><script src="polyfills/core-js.271cd0c5fd1a210a2c430d4d42c6ea72.js" nomodule=""></script><script>!function(){function e(e,o){return new Promise((function(t,n){document.head.appendChild(Object.assign(document.createElement("script"),{src:e,onload:t,onerror:n},o?{type:"module"}:void 0))}))}var o=[];function t(){window.importShim("./app.js")}"foo"in window&&o.push(e("polyfills/custom-a.612310cce7c28a680112cc9eff6ef77c.js",!1)),"fetch"in window||o.push(e("polyfills/fetch.a1ad5fb96dc0cb61b9454244c9bd7fe6.js",!1)),"noModule"in HTMLScriptElement.prototype&&!("importShim"in window)&&o.push(e("polyfills/dynamic-import.991be47e17117abb5eb15f5254ad3869.js",!1)),(!("attachShadow"in Element.prototype)||!("getRootNode"in Element.prototype)||window.ShadyDOM&&window.ShadyDOM.force)&&o.push(e("polyfills/webcomponents.f2956532e796a677ef87a9cb17f5f9fc.js",!1)),!("noModule"in HTMLScriptElement.prototype)&&"getRootNode"in Element.prototype&&o.push(e("polyfills/custom-elements-es5-adapter.84b300ee818dce8b351c7cc7c100bcf7.js",!1)),o.length?Promise.all(o).then(t):t()}()</script></body></html>
```

**Review by stephenwade — CHANGES_REQUESTED** [review]
If you change these two core-js hashes back, this should pass CI.

**Comment by stephenwade** [comment]
Ugh, it looks like the suggested changes added newlines to the end of the test snapshots, and now the test is complaining about those. So I can't help with suggested changes.

I suggest you reset the branch back to 60ccd55017c0df4aee04c042ee561f1f85ce3016, and then manually edit the affected snapshot files. These are under the packages/building-utils/test-node/index-html/snapshots/create-index-html/ folder:
- dynamic-import.html
- module-system.html
- polyfill-multiple-legacy.html
- polyfill-multiple.html
- polyfill-wc.html

Just change this in each one, commit that, and push again:
```diff
-<link rel="preload" href="app.js" as="script" crossorigin="anonymous">
+<link rel="modulepreload" href="app.js" as="script">
```

**Comment by soelen** [comment]
@stephenwade thanks for the help! Even thou changing certain files manually did not fully the trick ( looks like some line breaks were just missing?) the checks went finally through.

Also thanks to @Westbrook for pointing indirectly out it could be something with the OS or rather my dev enviroment. I fully cleaned and reinstalled the nodejs enviroment to make `test:update-snapshots` work correctly.

Furthermore I have to apologize since this has been a bit time consuming to check out this PR for some people; or so I believe at least. Thought I could finally contribute to the open-wc project but obviously I still lack of experience and knowledge. I hope next time when I create a PR something like this won't happen again : )

**Review by Westbrook — APPROVED** [review]
LGTM! Thanks for working through all the difficulties on this one @soelen

**Pull request merged**